src/System.Text.Json/tests/JsonObjectTests.cs

```diff
@@ -142,7 +142,7 @@ public static void TestDateTime()
         [Fact]
         public static void TestDateTimeOffset()
         {
-            DateTimeOffset dateTimeOffset = new DateTime(DateTime.MinValue.Ticks);
+            DateTimeOffset dateTimeOffset = new DateTime(DateTime.MinValue.Ticks, DateTimeKind.Utc);
```

Comment:
There are a few tests within S.T.Json.Tests that new up a `DateTime` (but don't specify `DateTimeKind`).

Do those need to be updated as well, or was this in particular the only concern?

For instance, what about these instances:
https://github.com/dotnet/corefx/blob/6257672b2172bf993568cf57ff5d2397bf1dcc7d/src/System.Text.Json/tests/JsonObjectTests.cs#L137
https://github.com/dotnet/corefx/blob/6257672b2172bf993568cf57ff5d2397bf1dcc7d/src/System.Text.Json/tests/Utf8JsonWriterTests.cs#L1010

Reply:
Apparently the exception is thrown when we cast `DateTime` to `DateTimeOffset` && `DateTime` is constructed with `new DateTime(DateTime.MinValue.Ticks)` or `new DateTime(0001, 1, 1)`:

```c#
using System;
static class C {
    static void Main() {
        DateTime dateTime = new DateTime(2019, 1, 1);                                                              
        DateTimeOffset dateTimeOffset = dateTime;   // this works

        DateTime dateTime2 = new DateTime(DateTime.MaxValue.Ticks);
        DateTimeOffset dateTimeOffset2 = dateTime2;   // this works

        DateTime dateTime3 = new DateTime(DateTime.MinValue.Ticks); // or new DateTime(0001, 1, 1)
        DateTimeOffset dateTimeOffset3 = dateTime3;   // this throws
    }
}
```